packages/gatsby/src/utils/__tests__/handle-flags.ts

```diff
@@ -14,44 +16,124 @@ describe(`handle flags`, () => {
       umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/28138`,
+      testFitness: (): fitnessEnum => true,
     },
     {
       name: `QUERY_ON_DEMAND`,
       env: `GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND`,
       command: `develop`,
+      telemetryId: `test`,
+      experimental: false,
       description: `Only run queries when needed instead of running all queries upfront. Speeds starting the develop server.`,
       umbrellaIssue: `https://github.com/gatsbyjs/gatsby/discussions/27620`,
-      noCi: true,
+      noCI: true,
+      testFitness: (): fitnessEnum => true,
     },
     {
       name: `ONLY_BUILDS`,
       env: `GATSBY_EXPERIMENTAL_BUILD_ME_FASTER`,
       command: `build`,
+      telemetryId: `test`,
+      experimental: false,
       description: `test`,
       umbrellaIssue: `test`,
+      testFitness: (): fitnessEnum => true,
     },
     {
       name: `ALL_COMMANDS`,
       env: `GATSBY_EXPERIMENTAL_SOMETHING_COOL`,
       command: `all`,
+      telemetryId: `test`,
+      experimental: false,
       description: `test`,
       umbrellaIssue: `test`,
+      testFitness: (): fitnessEnum => true,
     },
     {
       name: `YET_ANOTHER`,
       env: `GATSBY_EXPERIMENTAL_SOMETHING_COOL2`,
       command: `all`,
+      telemetryId: `test`,
+      experimental: false,
+      description: `test`,
+      umbrellaIssue: `test`,
+      testFitness: (): fitnessEnum => true,
+    },
+    {
+      name: `PARTIAL_RELEASE`,
+      env: `GATSBY_READY_TO_GO`,
+      command: `all`,
+      telemetryId: `test`,
+      experimental: false,
       description: `test`,
       umbrellaIssue: `test`,
+      testFitness: (flag): fitnessEnum => {
+        if (sampleSiteForExperiment(flag.name, 100)) {
+          return `OPT_IN`
+        } else {
+          return false
+        }
+      },
+    },
+    {
+      name: `PARTIAL_RELEASE_ONLY_VERY_OLD_LODASH`,
+      env: `GATSBY_READY_TO_GO_LODASH`,
+      command: `all`,
+      telemetryId: `test`,
+      experimental: false,
+      description: `test`,
+      umbrellaIssue: `test`,
+      testFitness: (flag): fitnessEnum => {
+        const semver = {
+          // Because of this, this flag will never show up
+          lodash: `<=3.9`,
+        }
+        if (
+          satisfiesSemvers(semver) &&
+          sampleSiteForExperiment(flag.name, 100)
+        ) {
+          return `OPT_IN`
+        } else {
+          return false
+        }
+      },
+    },
+    {
+      name: `PARTIAL_RELEASE_ONLY_NEW_LODASH`,
+      env: `GATSBY_READY_TO_GO_NEW_LODASH`,
+      command: `all`,
+      description: `test`,
+      umbrellaIssue: `test`,
+      telemetryId: `test`,
+      experimental: false,
+      testFitness: (flag): fitnessEnum => {
+        const semver = {
+          // Because of this, this flag will never show up
+          lodash: `>=4.9`,
+        }
+        if (
+          satisfiesSemvers(semver) &&
```

Comment:
how semver filtering would work — in a real flag, we'd generally also return `true` if it passes semver but doesn't pass the opt-in